### ISSUE TITLE
jQuery vs Backbone

### ISSUE THREAD

**Issue by mauryaratan** [kadamwhite/wordbone-pressback#4]
Considering Art gallery from the demo. Are there any direct advantage of using backbone.js over jQuery, even when you have to write more lines of code?

_On a note, the link to github is leading to 404 here - http://kadamwhite.github.io/talks/2014/backbone-wordpress-wpsessions/#/50_

**Comment by kadamwhite** [comment]
Thanks for the catch, I've fixed the link.

In this case, if this was all the code I was going to write, then I might not use Backbone here—it's a bit of a contrived example because the logic isn't very complicated. I chose it mainly because there's a simple model and a clear relationship between the model and several views.

On the other hand, I might not stop here: I might go on to remove the PHP-rendered `data-` attributes entirely, in favor of fetching information for those posts as-needed from an API endpoint. If I did that, I might only leave a `data-post-id` attribute, and use that to create an artwork model that I then `.fetch()` to retrieve the data. Or perhaps I want to make a series of widgets that each need to update to fetch & display related works—by having this be structured with Backbone, I have a clear way to create more areas of interactivity that all relate somehow to the selection.

It's definitely a judgement call, in that regard. I've heard a rule of thumb that when your code is big enough to start splitting it up into different files, it may be time to evaluate adding Backbone or another JavaScript framework. On the other hand, if you know you're going to be building a lot of complex single-page app-style interaction from the get-go, it'd be a no-brainer just to start out with Backbone. It's really going to depend on the needs of your specific webpage or application.
